shared/reducers/chat2.tsx

```diff
@@ -559,6 +559,21 @@ const reducer = Container.makeReducer<Actions, Types.State>(initialState, {
       orangeLineMap.delete(action.payload.conversationIDKey)
     }
   },
+  [Chat2Gen.markAsUnread]: (draftState, action) => {
+    const {conversationIDKey, readMsgID} = action.payload
+    const {markedAsUnreadMap, metaMap} = draftState
+    markedAsUnreadMap.set(conversationIDKey, true)
+    let old = metaMap.get(conversationIDKey)
+    if (old) {
+      old.readMsgID = readMsgID
```

Comment:
do we need to do it this way vs just getting the meta from the service? My preference is that the UI is dumber, its simpler if we can just 'update' ourselves instead of duplicating the logic on the go side